core/jobs/batch_jobs/exp_validation_jobs.py

```diff
@@ -0,0 +1,1197 @@
+
+        Returns:
+            drag_drop_interaction_values: List[str].
+            Invalid interaction values.
+        """
+        drag_drop_interaction_values = []
+
+        answer_groups = state.interaction.answer_groups
+        for answer_group in answer_groups:
+            answer_group_index = str(answer_groups.index(answer_group))
+            # Validates various value of DragAndDropSortInput interaction.
+            multi_item_value = (
+                state.interaction.customization_args
+                ['allowMultipleItemsInSamePosition'].value)
+            if not multi_item_value:
+                for rule_spec in answer_group.rule_specs:
+                    rule_spec_index = str(answer_group.rule_specs.index(
+                        rule_spec))
+                    # Validates multi items in same place iff setting on.
+                    for ele in rule_spec.inputs['x']:
+                        if len(ele) > 1:
+                            drag_drop_interaction_values.append(
+                                f'The rule {rule_spec_index} of '
+                                f'answer group {answer_group_index} '
+                                f'have multiple items at same place '
+                                f'when multiple items in same '
+                                f'position settings is turned off.'
+                            )
+
+                    # Validates == +/- 1 no option if multi item set off.
+                    if (
+                        rule_spec.rule_type ==
+                        'IsEqualToOrderingWithOneItemAtIncorrectPosition'
+                    ):
+                        drag_drop_interaction_values.append(
+                            f'The rule {rule_spec_index} '
+                            f'of answer group {answer_group_index} '
+                            f'having rule type - IsEqualToOrderingWith'
+                            f'OneItemAtIncorrectPosition should not '
+                            f'be there when the '
+                            f'multiple items in same position '
+                            f'setting is turned off.'
+                        )
+
+                    # Validates for a < b, a should not be the same as b.
+                    if (
+                        rule_spec.rule_type ==
+                        'HasElementXBeforeElementY'
+                    ):
+                        if (
+                            rule_spec.inputs['x'] == rule_spec.inputs['y']
+                        ):
+                            drag_drop_interaction_values.append(
+                                f'The rule {rule_spec_index} of '
+                                f'answer group {answer_group_index} '
+                                f'the value 1 and value 2 cannot be '
+                                f'same when rule type is '
+                                f'HasElementXBeforeElementY'
+                            )
```

Comment:
Why is this only checked for the not multi options?

Reply:
I have edited it now.